EveryDream_Tools_4_colab.ipynb

```diff
@@ -0,0 +1,561 @@
+        "--replace \"bob smith\""
+      ],
+      "metadata": {
+        "id": "6Y1md3OHAvhw"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Now we can chose to create text files based on our file names, this is usefull for images with very long discriptions or tag list, windows has a limit of 256 characters, and files will not transfer correctly to a windows program if they are longer, moving these files in a zip is fine however and causes no issues\n"
+      ],
+      "metadata": {
+        "id": "W0MspWmXJQuc"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "!python scripts/createtxtfromfilename.py"
+      ],
+      "metadata": {
+        "id": "BpvenvyQJr9b"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Compress our images "
+      ],
+      "metadata": {
+        "id": "boVkDsiWJ_-P"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "!python scripts/compress_img.py \\\n",
+        "--img_dir output \\\n",
+        "--out_dir output/compressed_images \\\n",
+        "--max_mp 1.5 \n",
+        "#--overwrite False \\\n",
+        "#--Quality 95  \\\n",
+        "#--noresize False \\\n",
+        "#--delete \\"
+      ],
+      "metadata": {
+        "id": "F6QYfylhKAII"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "HBrWnu1C_lN9"
+      },
+      "source": [
+        "## Download your DataSet from EveryDream/output\n",
+        "\n",
+        "If you're on a colab you can use the cell below to push your output to your Gdrive."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "ldW2sDLcAVgz"
+      },
+      "outputs": [],
+      "source": [
+        "\n",
+        "!mkdir /content/drive/MyDrive/Auto_Data_sets\n",
+        "!cp -r output/ /content/drive/MyDrive/Auto_Data_sets"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "B-HFqbP4AVgz"
+      },
+      "source": [
+        "## If not on colab/gdrive, the following will zip up your files for extraction\n",
+        "\n",
+        "You'll still need to use your runtime's own download feature to download the zip.\n",
+        "\n",
+        "![output zip](https://github.com/victorchall/EveryDream/blob/main/demo/output_zip.png?raw=1)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "SVa80mrKAVg0"
+      },
+      "outputs": [],
+      "source": [
+        "import patoolib\n",
+        "from google.colab import files\n",
+        "import os\n",
+        "\n",
+        "Zip_Location = \"/content/drive/MyDrive/output\" #@param {type:\"string\"}\n",
```

Comment:
wonder if this should be a dropdown to select other folders, it's very confusing on where stuff ends up when you run this various set of different things.  Caption.py will just make .txt files next to the input images by default, so /workspace/everydream2trainer/input is where everything would be.

Reply:
well if you point it to a Gdrive directory it writes the txt into the same folder as where the pictures already exist, this whole cell will need to be reworked now that its using ed2 as a base i think. since colab stores to everydreamlogs maybe there should be a subfolder in there for Gdrive users. as far as the above zip location, it can probably just get changed to the directory variable we use earlier, think the main issue is that old scripts output to output  but new one does to the input, so now that caption.py seems to work well ill get all these to start saving to similar locations and edit the markdowns as you have mentioned to be inline with the colab trainer, thanks for your time samich king!